lib/host.js

```diff
@@ -266,16 +271,21 @@ Host.prototype._checkPoolState = function () {
   if (this.pool.isClosing()) {
     return;
   }
+
   if (this.pool.connections.length < this.pool.coreConnectionsLength) {
-    // the pool still needs to grow
+    // the pool needs to grow / reconnect
     if (!this.pool.hasScheduledNewConnection()) {
       this.reconnectionDelay = this.reconnectionSchedule.next().value;
       this.pool.scheduleNewConnectionAttempt(this.reconnectionDelay);
     }
   }
-  if (this._distance !== types.distance.ignored &&
-      this.pool.connections.length === 0 &&
-      this.pool.coreConnectionsLength > 0) {
+
+  if (
+    this._distance !== types.distance.ignored &&
+    this.pool.connections.length === 0 &&
+    this.pool.coreConnectionsLength > 0
+  ) {
```

Comment:
I would not change the code style in this `if`.

Reply:
I'm not a big fan of any multiline if code style, I've changed to use airbnb's convention: https://github.com/airbnb/javascript/issues/1380

I think the clearest way would be to assign it to a variable, i.e., `shouldHaveConnections`.